### PR TITLE
Replaced several attribute calls with single Files.readAttributes call

### PR DESCRIPTION
**Pull request by StefanSpieker**
Replaced several attribute calls with single `Files.readAttributes` call. This might improve performance.

### Proposed changelog entries

* N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content-Security-Policy directives (see [documentation on jenkins.io](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6981"></a>

### CODE REVIEW

File: core/src/main/java/hudson/util/io/ZipArchiver.java
Comment (basil) [inline core/src/main/java/hudson/util/io/ZipArchiver.java:86]: Careful! `File#toPath` can throw `InvalidPathException`. If callers aren't already handling `InvalidPathException` (which is often the case for methods that rethrow `IOException` rather than catching and handling `IOException | InvalidPathException `), better to use `Util.fileToPath`.

File: core/src/main/java/hudson/util/io/TarArchiver.java
Comment (basil) [inline core/src/main/java/hudson/util/io/TarArchiver.java:87]: Careful! `File#toPath` can throw `InvalidPathException`. If callers aren't already handling `InvalidPathException` (which is often the case for methods that rethrow `IOException` rather than catching and handling `IOException | InvalidPathException `), better to use `Util.fileToPath`.

File: core/src/main/java/hudson/util/io/TarArchiver.java
Comment (basil) [inline core/src/main/java/hudson/util/io/TarArchiver.java:37]: I think this blank line will trip up the linters. Better to run `mvn clean verify -DskipTests` locally before posting a PR to minimize turnaround time over stuff like this.

**Review by basil — COMMENTED** [review]
A bit of premature optimization, but should be safe once the review comments are addressed.

**Comment by basil** [comment]
Note that this metadata will usually be in the page cache, so this won't have much of a performance impact in practice except on NFS.

File: core/src/main/java/hudson/util/io/ZipArchiver.java
Comment (StefanSpieker) [reply core/src/main/java/hudson/util/io/ZipArchiver.java]: Thanks for the tip, adjusted as suggested.

File: core/src/main/java/hudson/util/io/TarArchiver.java
Comment (StefanSpieker) [reply core/src/main/java/hudson/util/io/TarArchiver.java]: Thanks for the tip, adjusted as suggested.

File: core/src/main/java/hudson/util/io/TarArchiver.java
Comment (StefanSpieker) [reply core/src/main/java/hudson/util/io/TarArchiver.java]: I should not forget to disable the `quick-build` ;)

**Review by basil — APPROVED** [review]
Premature optimization, but why not?

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by NotMyFault — APPROVED** [review]

**Pull request merged**
